detection_rules/version_lock.py

```diff
@@ -242,13 +244,26 @@ def add_changes(r, *msg):
                 elif min_stack < latest_locked_stack_version:
                     route = 'C'
                     # 4) on an old stack, after a breaking change has been made (updated fork)
+                    # 5) on an old stack, locking in an old breaking change
                     assert str(min_stack) in existing_rule_lock.get("previous", {}), \
                         f"Expected {rule.id} @ v{min_stack} in the rule lock"
 
-                    # TODO: Figure out whether we support locking old versions and if we want to
+                    # Support locking old versions if we have buffer space in between
                     #       "leave room" by skipping versions when breaking changes are made.
                     #       We can still inspect the version lock manually after locks are made,
                     #       since it's a good summary of everything that happens
+                    new_min_version = max([x['version'] for x in existing_rule_lock['previous'].values()])
+                    new_max_version = existing_rule_lock["version"]
+
+                    # leave half the space available since we're updating an old previous version
+                    # or inserting an old breaking change on an old stack
+                    new_version = (new_max_version - new_min_version) / 2
+                    if new_version <= 1:
+                        # we're out of buffer space
+                        raise ValueError(f'Out of buffer space on {rule.id} '
```

Comment:
Should we include verbiage about what to do if out of buffer space for a rule. I think the only solution is to deprecate and create a new one, based on our conversation.